apps/system/test/apps/wifi-direct/js/wifip2p_listview.js

```diff
@@ -0,0 +1,58 @@
+/* -*- Mode: js; js-indent-level: 2; indent-tabs-mode: nil -*- */
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+/* global dump */
+
+'use strict';
+
+(function(exports) {
+  function debug(s) {
+    dump('Wifi Direct Test: ' + s);
+  }
+
+  function WifiP2pPeerListView(aListView) {
+    this.onPeerClicked = function() {};
+    this.listView = aListView;
+  }
+
+  WifiP2pPeerListView.prototype = {
+    clear: function() {
+      // remove all items except the text expl. and the "search again" button
+      var peerItems = this.listView.querySelectorAll('li');
+      var len = peerItems.length;
+      for (var i = len - 1; i >= 0; i--) {
+        this.listView.removeChild(peerItems[i]);
+      }
+    },
+
+    setPeerList: function(aPeerList) {
+      this.clear();
+      debug('PeerList to set: ' + JSON.stringify(aPeerList));
+      for (var i = 0; i < aPeerList.length; i++) {
+        var listItem = this._newPeerListItem(aPeerList[i], this.onPeerClicked);
```

Comment:
When you pass `this.onPeerClicked` to the function you actually save the value to a variable, which makes you to set `this.onPeerClicked` before setting peer list to make things work. I would suggest to call to `this.onPeerClicked` directly in the click handler.